libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -990,6 +991,58 @@ impl NodeAPI for Greenlight {
         debug!("send_custom_message returned status {:?}", resp.status);
         Ok(())
     }
+
+    async fn fetch_invoice(&self, req: FetchInvoiceRequest) -> NodeResult<FetchInvoiceResponse> {
+        // Parse the offer locally, to avoid any unnecessary calls to the recipient
+        if let Err(_parse_error) = req.offer.parse::<Offer>() {
+            return Err(NodeError::InvalidOffer(anyhow!("Invalid offer")));
+        };
+
+        // Fetch the invoice from the recipient's node
+        let mut client = self.get_node_client().await?;
+        let response = client
+            .fetch_invoice(Into::<cln::FetchinvoiceRequest>::into(req))
+            .await?
+            .into_inner();
+
+        Ok(FetchInvoiceResponse {
+            bolt12: response.invoice,
+            new_amount_msat: match response.changes {
+                Some(changes) => changes.amount_msat.map(|amount| amount.msat),
```

Comment:
Do we need to check if other changes exists and return error? My concern is that it seems we swallow these changes without any indication for the user.

Reply:
For basic support, currently not. Changes in description/vendor can be ignored while a change in the amount could mean the user paying more than the expects, thus an error has to be thrown